seqr/views/apis/data_manager_api.py

```diff
@@ -387,14 +387,81 @@ def load_rna_seq_sample_data(request, sample_guid):
-        'dbEntity': model_cls.__name__, 'numEntities': len(models), 'parentEntityIds': [sample_guid], 'updateType': 'bulk_create',
-    })
+    model_cls.bulk_create(request.user, [model_cls(sample=sample, **data) for data in data_by_gene.values()])
 
     return create_json_response({'success': True})
 
 
+@data_manager_required
+def load_phenotype_prioritization_data(request):
+    request_json = json.loads(request.body)
+
+    file_path = request_json['file']
+
+    try:
+        tool, data_by_project_indiv_id = load_phenotype_prioritization_data_file(file_path)
+    except ValueError as e:
+        return create_json_response({'error': str(e)}, status=400)
+
+    info = [f'Loaded {tool.title()} data from {file_path}']
+    logger.info(f'Loading {tool.title()} data from {file_path}', request.user)
+
+    internal_projects = get_internal_projects().filter(name__in=data_by_project_indiv_id)
+    projects_by_name = {p_name: [project for project in internal_projects if project.name == p_name]
+                       for p_name in data_by_project_indiv_id.keys()}
+    missing_projects = [p_name for p_name, projects in projects_by_name.items() if len(projects) == 0]
+    missing_info = f"Project {', '.join(missing_projects)} not found. " if missing_projects else ''
+    conflict_projects = [p_name for p_name, projects in projects_by_name.items() if len(projects) > 1]
+    conflict_info = f"Projects with conflict name(s) {', '.join(conflict_projects)}." if conflict_projects else ''
+
+    if missing_info or conflict_info:
+        return create_json_response({'error': missing_info + conflict_info}, status=400)
+
+    all_records = []
+    to_delete = PhenotypePrioritization.objects.none()
+    error = None
+    for project_name, records_by_indiv in data_by_project_indiv_id.items():
+        indivs = Individual.objects.filter(family__project=projects_by_name[project_name][0],
+                                           individual_id__in=records_by_indiv.keys())
+        existing_indivs_by_id = {ind.individual_id: ind for ind in indivs}
+
+        missing_individuals = set(records_by_indiv.keys()) - set(existing_indivs_by_id.keys())
+        if missing_individuals:
+            error = f"Can't find individuals {', '.join(sorted(list(missing_individuals)))}"
+            break
+        indiv_records = []
+        for sample_id, records in records_by_indiv.items():
+            for rec in records:
+                rec['individual'] = existing_indivs_by_id[sample_id]
+                indiv_records.append(rec)
+
+        exist_records = PhenotypePrioritization.objects.filter(tool=tool, individual__in=indivs)
+
+        delete_info = f'deleted {len(exist_records)} record(s), ' if exist_records else ''
+        info.append(f'Project {project_name}: {delete_info}loaded {len(indiv_records)} record(s)')
+
+        logger.info(
+            f'Project {project_name}: deleting {len(exist_records)} record(s), loading {len(indiv_records)} record(s)',
```

Comment:
If you are going to log and put in the info field the exact same message except one is "loading" and one is "loaded" just log the same message as the info, the complexity is not worth it

Reply:
Could it be misleading to say `loaded` but not loaded because of missing individuals later on?

Reply:
then move the log message to a place where it is true. Again, its really not important to have logs and info messages line up - honestly since we log the bulk creates and deletes its probably not necessary to log these summary stats at all